extra_data/components.py

```diff
@@ -1102,6 +1191,12 @@
 
         return arr
 
+
+class XtdfMaskedKeyData(DetectorMaskedKeyData, XtdfImageMultimodKeyData):
```

Comment:
I believe this is a false positive caused by CodeQL not fully understanding `super()` with multiple inheritance. The `__init__` method is inherited from `DetectorMaskedKeyData`, which does call `super().__init__(...)`. That calls the next `__init__` in the method resolution order - not just the parent class - so when it's creating an `XtdfMaskedKeyData`, it will find `XtdfImageMultimodKeyData.__init__()`.

```
In [2]: XtdfMaskedKeyData.__mro__
Out[2]: 
(extra_data.components.XtdfMaskedKeyData,
 extra_data.components.DetectorMaskedKeyData,
 extra_data.components.XtdfImageMultimodKeyData,
 extra_data.components.MultimodKeyData,
 object)
```

I usually avoid multiple inheritance because even if it works, it's a pain to think about stuff like this. In this instance, I started writing the masked KeyData classes as wrappers around the existing classes (has-a relationships rather than is-a), but it was going to be several times more code to provide the same methods and attributes again. So I decided using inheritance was the lesser of two evils.